generator/templates/client/parameter.gotmpl

```diff
@@ -104,6 +117,16 @@ func ({{ .ReceiverName }} *{{ pascalize .Name }}Params) SetContext(ctx context.C
   {{ .ReceiverName }}.Context = ctx
 }
 
+// WithHTTPClient adds the HTTPClient to the {{ humanize .Name }} params
+func ({{ .ReceiverName }} *{{ pascalize .Name }}Params) WithHTTPClient(client *http.Client) *{{ pascalize .Name }}Params {
```

Comment:
@casualjim isn't that this one: https://github.com/go-swagger/go-swagger/pull/849/commits/e50ccb700cd77d1a0fdbc7a7ce6d5e29a946a4fc#diff-643df0026bfd7f1638796085caeb52f5R127?